test/e2e/common.go

```diff
@@ -0,0 +1,154 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	bmov1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LoadImageBehavior indicates the behavior when loading an image.
+type LoadImageBehavior string
+
+const (
+	// MustLoadImage causes a load operation to fail if the image cannot be
+	// loaded.
+	MustLoadImage LoadImageBehavior = "mustLoad"
+
+	// TryLoadImage causes any errors that occur when loading an image to be
+	// ignored.
+	TryLoadImage LoadImageBehavior = "tryLoad"
+)
+
+// Config defines the configuration of an e2e test environment.
+type Config struct {
+	// Images is a list of container images to load into the Kind cluster.
```

Comment:
I'm not following. This is specifically only used with the automatically created Kind cluster. If a pre-existing cluster is used instead (e.g. minikube when running `ci-e2e.sh`) then no images will be loaded because there is no way for the code to know how to do it. Not all clusters has a way to load images.
Maybe I need to update this comment to make it clearer that it is only relevant when *not* using an existing cluster?